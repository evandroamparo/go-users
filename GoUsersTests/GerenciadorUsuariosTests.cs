using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using GoUsers.Model;
using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace GoUsers.Tests
{
  [TestClass]
  public class GerenciadorUsuariosTests
  {
    private Usuario usuario;
    private GerenciadorUsuarios gerenciadorUsuarios;
    private string arquivoTeste;

    [TestInitialize]
    public void Setup()
    {
      usuario = new Usuario("admin", "senha", true);
      arquivoTeste = "usuarios.txt";
      if (File.Exists(arquivoTeste))
      {
        File.Delete(arquivoTeste);
      }
      gerenciadorUsuarios = new GerenciadorUsuarios(arquivoTeste);
    }

    [TestMethod]
    [ExpectedException(typeof(UsuarioDuplicadoException))]
    public void LoginNaoPodeSerDuplicado()
    {
      gerenciadorUsuarios.AdicionarUsuario(usuario);
      var usuarioDuplicado = new Usuario("ADMIN", "teste");
      gerenciadorUsuarios.AdicionarUsuario(usuarioDuplicado);
    }

    [TestMethod]
    public void UsuarioAdicionadoDeveConstarNaLista()
    {
      gerenciadorUsuarios.AdicionarUsuario(usuario);
      CollectionAssert.Contains(gerenciadorUsuarios.Usuarios, usuario);
    }

    [TestMethod]
    public void UsuarioExcluidoNaoDeveConstarNaLista()
    {
      gerenciadorUsuarios.AdicionarUsuario(usuario);
      gerenciadorUsuarios.ExcluirUsuario(usuario);
      CollectionAssert.DoesNotContain(gerenciadorUsuarios.Usuarios, usuario);
    }

    [TestMethod]
    public void ArquivoInexistenteDeveInicializarListaVazia()
    {
      gerenciadorUsuarios = new GerenciadorUsuarios("nome de arquivo inexistente");
      Assert.AreEqual(0, gerenciadorUsuarios.Usuarios.Count);
    }

    [TestMethod]
    [UseReporter(typeof(DiffReporter), typeof(FrameworkAssertReporter))]
    public void ArquivoExistenteDeveInicializarListaPreenchida()
    {
      gerenciadorUsuarios = new GerenciadorUsuarios("exemplos/usuarios.txt");
      Approvals.Verify(gerenciadorUsuarios);
    }

    [TestMethod]
    [UseReporter(typeof(DiffReporter))]
    public void NovosUsuariosDevemSerPersistidosNoArquivo()
    {
      var novoUsuario = new Usuario("NovoUsuario", "123456");
      gerenciadorUsuarios.AdicionarUsuario(novoUsuario);
      gerenciadorUsuarios = new GerenciadorUsuarios(arquivoTeste);
      Approvals.Verify(gerenciadorUsuarios);
    }

    [TestMethod]
    public void UsuariosExcluidosDevemSerExcluidosDoArquivo()
    {
      gerenciadorUsuarios.AdicionarUsuario(usuario);
      gerenciadorUsuarios.ExcluirUsuario(usuario);
      gerenciadorUsuarios = new GerenciadorUsuarios(arquivoTeste);
      Usuario usuarioRecuperado = gerenciadorUsuarios.RecuperarUsuario("admin");
      Assert.IsNull(usuarioRecuperado);
    }
  }
}
