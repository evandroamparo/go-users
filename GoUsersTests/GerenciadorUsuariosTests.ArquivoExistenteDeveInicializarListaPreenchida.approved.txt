admin:ThmbShxAtJepX80c2JY1FzOEmUk=
123456:fEqNCco3Yq9h5ZUglD3CZJT4lBs=
